src/hotspot/share/opto/graphKit.cpp

```diff
@@ -619,6 +619,40 @@ void GraphKit::builtin_throw(Deoptimization::DeoptReason reason, Node* arg) {
     }
   }
 
+  if (treat_throw_as_hot && OptimizeImplicitExceptions) {
+    ciInstanceKlass* ex_ciInstKlass = env()->exception_instanceKlass_for_reason(reason);
+    if (ex_ciInstKlass != NULL) {
+      const TypeKlassPtr *ex_type = TypeKlassPtr::make(ex_ciInstKlass);
+      kill_dead_locals();
+      Node* ex_node = new_instance(makecon(ex_type), NULL, NULL, true);
+      set_argument(0, ex_node);
+      ciMethod* init = ex_ciInstKlass->find_method(ciSymbol::make("<init>"), ciSymbol::make("()V"));
```

Comment:
I know that all exceptions classes have such constructor but in general you need to check for `nullptr`. I think it could be moved before check at line 624.